httpserver/service_node.cpp

```diff
@@ -397,50 +400,217 @@ void ServiceNode::attach_signature(std::shared_ptr<request_t>& request,
     request->set(LOKI_SENDER_SNODE_PUBKEY_HEADER, stripped);
 }
 
-void ServiceNode::perform_peer_test() {
+void abort_if_integration_test() {
+#ifdef INTEGRATION_TEST
+    abort();
+#endif
+}
 
-    // 1. Select the tester/testee pair
+void ServiceNode::send_message_test_req(const sn_record_t& testee,
+                                        const Item& item) {
+
+    auto callback = [testee, item,
+                     height = this->block_height_](sn_response_t&& res) {
+        if (res.error_code == SNodeError::NO_ERROR && res.body) {
+            if (*res.body == item.data) {
+                BOOST_LOG_TRIVIAL(debug)
+                    << "Message test is successful for: " << testee
+                    << " at height: " << height;
+            } else {
+                BOOST_LOG_TRIVIAL(warning)
+                    << "Test answer doesn't match for: " << testee
+                    << " at height: " << height;
+
+#ifdef INTEGRATION_TEST
+                BOOST_LOG_TRIVIAL(warning)
+                    << "got: " << *res.body << " expected: " << item.data;
+#endif
+                abort_if_integration_test();
+            }
+        } else {
+            BOOST_LOG_TRIVIAL(error)
+                << "Failed to send a message test request to snode: " << testee;
 
-    std::vector<sn_record_t> members_pk = swarm_->other_nodes();
+            abort_if_integration_test();
+        }
+    };
 
-    members_pk.push_back(our_address_);
+    nlohmann::json json_body;
 
-    std::sort(members_pk.begin(), members_pk.end());
+    json_body["height"] = block_height_;
+    json_body["hash"] = item.hash;
 
-    if (members_pk.size() < 2) {
+    auto req = std::make_shared<request_t>();
+    req->method(http::verb::post);
+    req->set(http::field::host, "service node");
+    req->body() = json_body.dump();
+    req->target("/msg_test");
+    req->prepare_payload();
+
+    make_http_request(ioc_, testee.address, testee.port, req, callback);
```

Comment:
as a TODO, we should disambiguate `make_post_request` (instantiate a request object) from `make_http_request` (starts sending that object).

Reply:
Added a TODO